### PR TITLE
add jenkins-playbook stack

### PR DESCRIPTION
**Pull request by franck-art**
Jenkins installation stack with playbook

### CODE REVIEW

**Comment by diranetafen** [comment]
Il faudrait qu'on revois ton organisation du code
car tu as créé des répertoires et fichiers non nécessaires

**Comment by franck-art** [comment]
add install_jenkins playbook, requirements.yml  and jenkins_VM.yml stack

File: ansible/roles/requirements.yml [hunk 1]
```diff
@@ -6,3 +6,12 @@
 
 # Install a role for gitlab/gitlab-runner
 - src: https://github.com/HP41/ansible-docker-gitlab.git 
+
+# install java
+
```
Comment (diranetafen) [inline ansible/roles/requirements.yml:11]: remove this empty line

File: ansible/roles/requirements.yml [hunk 2]
```diff
@@ -6,3 +6,12 @@
...
+- src: geerlingguy.java
+
+# install jenkins
+
```
Comment (diranetafen) [inline ansible/roles/requirements.yml:15]: remove this empty line

File: stack/jenkins_VM.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            git clone https://github.com/diranetafen/cursus-devops.git
+            cd cursus-devops/ansible 
+            ansible-galaxy role install -r roles/requirements.yml
+            ansible-playbook install_jenkins.yml
+           
+            # add the user centos in the docker group
+            sudo usermod -aG docker centos
+
+            # permission to execute docker commands with the socket
+            sudo chmod 777 /var/run/docker.sock
```
Comment (diranetafen) [inline stack/jenkins_VM.yml:73]: WARNING, i cannot authorize this
Why you need to make this ? it's very dangerous

File: ansible/install_jenkins.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  pre_tasks:
+  - name: update all packages 
+    yum:
+      name: '*'
+      update_only: yes
+      update_cache: yes
+      state: latest
+
+  - name: install  nano
```
Comment (diranetafen) [inline ansible/install_jenkins.yml:21]: on server, we don't need nano, the server is not used for editing purpose !

File: ansible/install_jenkins.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+- hosts: localhost
+  become: true
+  gather_facts: true
+  
+  vars:
+    jenkins_hostname: localhost
+    java_packages:
+      - java-1.8.0-openjdk
+
```
Comment (diranetafen) [inline ansible/install_jenkins.yml:9]: add a variable for plugin install even if it's empty : jenkins_plugins: []

File: stack/jenkins_VM.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (diranetafen) [inline stack/jenkins_VM.yml:1]: change the name of the file as jenkins_vm and not jenkins_VM
Capital letter is not needed for those files

File: ansible/roles/requirements.yml (hunk 1 above)
Comment (franck-art) [reply ansible/roles/requirements.yml]: corrected remarks

File: ansible/roles/requirements.yml (hunk 2 above)
Comment (franck-art) [reply ansible/roles/requirements.yml]: corrected remarks

File: ansible/install_jenkins.yml (hunk 4 above)
Comment (franck-art) [reply ansible/install_jenkins.yml]: corrected remarks

File: ansible/install_jenkins.yml (hunk 5 above)
Comment (franck-art) [reply ansible/install_jenkins.yml:11]: corrected remarks

File: stack/jenkins_VM.yml (hunk 6 above)
Comment (franck-art) [reply stack/jenkins_VM.yml]: corrected remarks

File: ansible/install_jenkins.yml (hunk 4 above)
Comment (franck-art) [reply ansible/install_jenkins.yml]: corrected remarks

File: ansible/install_jenkins.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+- hosts: localhost
+  become: true
+  gather_facts: true
```
Comment (diranetafen) [inline ansible/install_jenkins.yml:3]: tu n'as pas besoin de cette ligne tu peux la supprimer

Comment (franck-art) [reply ansible/install_jenkins.yml]: c'est reglé

**Pull request merged**